### PR TITLE
ci: Add automated release workflow

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# Trigger the workflow when:
+on:
+  # A push occurs to one of the matched tags.
+  push:
+    tags:
+      # Pattern that roughly matches Ledger JS' version tags.
+      # For more details on GitHub Actions' pattern match syntax, see:
+      # https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags.
+      - 'v[0-9]+.[0-9]+*'
```
Comment (ptrus) [inline .github/workflows/release.yml:12]: Maybe a short note about the release process in the readme would be useful: e.g. "to publish a new version of the package bump the version and push a `v<version>` tag."

**Review by ptrus — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (kostko) [reply .github/workflows/release.yml:12]: Added a simple release process doc, we'll improve it once we finish #12.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          # Check out pull request's HEAD commit instead of the merge commit to
+          # work-around an issue where a wrong commit is being checked out.
+          # For more details, see:
+          # https://github.com/actions/checkout/issues/299.
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install needed system packages
```
Comment (tjanez) [inline .github/workflows/release.yml:27]: Nitpick: Add a new line between the 2 steps here or remove them everywhere else

**Review by tjanez — APPROVED** [review]
Looks good!

**Pull request merged**